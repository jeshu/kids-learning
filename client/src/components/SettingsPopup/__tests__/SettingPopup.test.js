import React from "react";
import { render,  screen, fireEvent } from "@testing-library/react";
import SettingPopup from "../";

describe('Test suit for Setting popup', () => {

    it('should not render in dom', () => {
        render(<SettingPopup show={false} />)
        expect(screen.queryByTestId('settings-popup')).toBeNull();
    })

    it('should render without error', () => {
        const { asFragment} = render(<SettingPopup show={true} />);
        expect(asFragment()).toMatchSnapshot();
        expect(screen.queryByTestId('settings-popup')).toBeInTheDocument();
    })
    
    it('should not be in the dom once close btn clicked', () => {
        const onClose = jest.fn();
        let showState = true;
        const {getByTestId, rerender} = render(<SettingPopup show={showState} onClose={onClose}  />);
        fireEvent.click(screen.queryByTestId('settings-close'));
        expect(onClose).toHaveBeenCalled();

    })
    
    it('should have dropdown to change music', () => {
        render(<SettingPopup show={true} />)
        expect(screen.queryByTestId('settings-music')).toBeInTheDocument();
    })
});