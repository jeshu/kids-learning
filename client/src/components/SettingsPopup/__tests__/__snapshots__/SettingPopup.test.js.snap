// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suit for Setting popup should render without error 1`] = `<DocumentFragment />`;
